core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -31,14 +31,46 @@ class ControllerContext {
   var epoch: Int = KafkaController.InitialControllerEpoch - 1
   var epochZkVersion: Int = KafkaController.InitialControllerEpochZkVersion - 1
   var allTopics: Set[String] = Set.empty
-  var partitionReplicaAssignment: mutable.Map[TopicPartition, Seq[Int]] = mutable.Map.empty
+  private var partitionReplicaAssignmentUnderlying: mutable.Map[String, mutable.Map[Int, Seq[Int]]] = mutable.Map.empty
   var partitionLeadershipInfo: mutable.Map[TopicPartition, LeaderIsrAndControllerEpoch] = mutable.Map.empty
   val partitionsBeingReassigned: mutable.Map[TopicPartition, ReassignedPartitionsContext] = mutable.Map.empty
   val replicasOnOfflineDirs: mutable.Map[Int, Set[TopicPartition]] = mutable.Map.empty
 
   private var liveBrokersUnderlying: Set[Broker] = Set.empty
   private var liveBrokerIdsUnderlying: Set[Int] = Set.empty
 
+  def partitionReplicaAssignment(topicPartition: TopicPartition) : Seq[Int] = {
+    partitionReplicaAssignmentUnderlying.getOrElse(topicPartition.topic, mutable.Map.empty)
+      .getOrElse(topicPartition.partition, Seq.empty)
+  }
+
+  def clearPartitionReplicaAssignment() = {
+    partitionReplicaAssignmentUnderlying = mutable.Map.empty
+  }
+
+  def updatePartitionReplicaAssignment(topicPartition: TopicPartition, newReplicas : Seq[Int]) = {
+    partitionReplicaAssignmentUnderlying.getOrElseUpdate(topicPartition.topic, mutable.Map.empty)
+      .put(topicPartition.partition, newReplicas)
+  }
+
+  def partitionReplicaAssignmentForTopic(topic : String) : mutable.Map[TopicPartition, Seq[Int]] = {
+    partitionReplicaAssignmentUnderlying.getOrElse(topic, mutable.Map.empty).map {
+      case (partition, replicas) => (new TopicPartition(topic, partition), replicas)
+    }
+  }
+
+  def removePartitionReplicaAssignmentForTopic(topic : String) = {
```

Comment:
It seems that this is the same method as removeTopic? The latter seems more complete.

Reply:
Removed this method and replaced it with removeTopic instead.